cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -292,118 +285,198 @@ fn storage_footprint_to_ledger_footprint(foot: &Footprint) -> Result<LedgerFootp
     })
 }
 
+fn finalize_transaction_data_and_min_fee(
+    pre_storage: &ledger_storage::LedgerStorage,
+    transaction_resources: &TransactionResources,
+    soroban_resources: SorobanResources,
+    rent_changes: &Vec<LedgerEntryRentChange>,
+    current_ledger_seq: u32,
+    bucket_list_size: u64,
+) -> Result<(SorobanTransactionData, i64), Box<dyn error::Error>> {
+    let (fee_configuration, rent_fee_configuration) =
+        get_fee_configurations(pre_storage, bucket_list_size)?;
+    let (non_refundable_fee, refundable_fee) =
+        compute_transaction_resource_fee(transaction_resources, &fee_configuration);
+    let rent_fee = compute_rent_fee(&rent_changes, &rent_fee_configuration, current_ledger_seq);
+    let transaction_data = SorobanTransactionData {
+        resources: soroban_resources,
+        refundable_fee: refundable_fee + rent_fee,
+        ext: ExtensionPoint::V0,
+    };
+    let res = (
+        transaction_data,
+        refundable_fee + non_refundable_fee + rent_fee,
+    );
+    Ok(res)
+}
+
 pub(crate) fn compute_bump_footprint_exp_transaction_data_and_min_fee(
     footprint: LedgerFootprint,
     ledgers_to_expire: u32,
-    snapshot_source: &ledger_storage::LedgerStorage,
+    ledger_storage: &ledger_storage::LedgerStorage,
     bucket_list_size: u64,
+    current_ledger_seq: u32,
 ) -> Result<(SorobanTransactionData, i64), Box<dyn error::Error>> {
+    // FIXME: we are querying the ledger entries from the DB twice (once here and again in calculate_unmodified_ledger_entry_bytes)
+    let mut rent_changes: Vec<LedgerEntryRentChange> = Vec::new();
+    for key in (&footprint).read_only.as_vec() {
+        let unmodified_entry = ledger_storage.get(key, false)?;
+        let (expiration_ledger, durability) =
+            get_bumpable_entry_expiration_and_durability(&unmodified_entry)?;
+        let new_expiration_ledger = current_ledger_seq + ledgers_to_expire;
+        if new_expiration_ledger <= expiration_ledger {
```

Comment:
You also need to check that entry is still alive (current ledger >= expiration ledger). It's not possible to bump an expired entry.

Reply:
That's ensured by the `includeExpired: false` parameter of `ledger_storage.get(key, false)` a few lines above

Reply:
Ok, sounds good.